packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,398 @@
+        return proposals[_proposalId].tally.approvals;
+    }
+
+    /// @notice Updates the minimal approval parameter.
+    /// @param _minApprovals The new minimal approval value.
+    function updateMinApprovals(uint256 _minApprovals)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _updateMinApprovals(_minApprovals);
+
+        emit MinApprovalUpdated({minApprovals: _minApprovals});
+    }
+
+    /// @notice Adds new members to the address list and updates the minimum approval parameter.
+    /// @param _members The addresses of the members to be added.
+    function addAddresses(address[] calldata _members)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _updateAddresslist(_members, true);
+
+        emit AddressesAdded({members: _members});
+    }
+
+    /// @notice Removes existing members from the address list. Previously, it checks if the new addresslist length at least as long as the minimum approvals parameter requires. Note that `minApprovals` is must be at least 1 so the address list cannot become empty.
+    /// @param _members The addresses of the members to be removed.
+    function removeAddresses(address[] calldata _members)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        // Check if the new addresslist has become shorter than the current minimum number of approvals required.
+        uint256 newAddresslistLength = addresslistLength(0) - _members.length;
+        if (newAddresslistLength < minApprovals_) {
+            revert MinApprovalsOutOfBounds({limit: newAddresslistLength, actual: minApprovals_});
+        }
+
+        _updateAddresslist(_members, false);
+
+        emit AddressesRemoved({members: _members});
+    }
+
+    /// @notice Creates a new majority voting proposal.
+    /// @param _metadata The metadata of the proposal.
+    /// @param _actions The actions that will be executed after the proposal passes.
+    /// @param _approve_ If true, the sender will approve the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the vote cast. The call does not revert if early execution is not possible.
+    /// @return id The ID of the proposal.
+    function createProposal(
+        bytes calldata _metadata,
+        IDAO.Action[] calldata _actions,
+        bool _approve_,
+        bool _tryExecution
+    ) external returns (uint256 id) {
+        uint64 snapshotBlock = getBlockNumber64() - 1;
+
+        if (!isListed(_msgSender(), snapshotBlock)) {
+            revert ProposalCreationForbidden(_msgSender());
+        }
+
+        id = _createProposal(_msgSender(), _metadata, _actions);
+
+        // Create the proposal
+        Proposal storage proposal_ = proposals[id];
+
+        proposal_.open = true;
+        proposal_.parameters.snapshotBlock = snapshotBlock;
+        proposal_.parameters.minApprovals = minApprovals_;
```

Comment:
It feels a bit weird to call "_createProposal" and immediately after... create it ourselves

Reply:
We will work on it in https://aragonassociation.atlassian.net/browse/APP-1447
so this comment can be resolved